File: bin/assets/core/client/scripts/ui/console.lua


local console_document
local console_visible = false

local function on_key_up(ev)
    -- keycode 44 is tilde, todo make a nice table for looking up by names
    if (ev.parameters.key_identifier == 44) then
        console_visible = not console_visible
        if(console_visible) then
            console_document:Show()
            local element = console_document:GetElementById("console-input")            
            element:Focus()
            mud.p_ticker_pause(true)
        else 
            console_document:Hide()
            mud.p_ticker_pause(false)
        end
    end
end

local function load(context)
    console_document = context:LoadDocument("console.rml")
    console_document:PushToBack() -- for debug ui
    context:AddEventListener('keyup', on_key_up, true)
end

return {
    load = load
}

File: script/source/client/ui/console.ts


let console_document: RmlElementDocument;
let console_visible = false

function on_key_up(ev: any) {
    // keycode 44 is tilde, todo make a nice table for looking up by names
    if (ev.parameters.key_identifier == 44) {
        console_visible = !console_visible
        if (console_visible) {
            console_document.Show()
            let element = console_document.GetElementById("console-input")
            element.Focus()
            mud.p_ticker_pause(true)
        }
        else {
            console_document.Hide()
            mud.p_ticker_pause(false)
        }

    }
}

function load(context: RmlContext) {
    console_document = context.LoadDocument("console.rml")
    console_document.PushToBack() // for debug ui
    context.AddEventListener('keyup', on_key_up, true)
}


export default {
    load: load
}